### PR TITLE
Issue 1270: Add Bazel dependency formatter for Maven artifacts

### LINKED ISSUE #1270: Add Bazel dependency formatter for Maven artifacts
# Task Description

We would like to add support for copy-pasteable dependencies for [Bazel](https://www.bazel.build/) to be used in the search engine. The format is outlined [here](https://docs.bazel.build/versions/master/be/workspace.html#maven_jar).

# Tasks

* [ ] Implement `BazelDependencyFormatter`.
* [ ] Implement a test case for it that checks things with different combinations of Maven artifact coordinates (see `MavenArtifactCoordinates` for more details) -- with, or without classifiers, different packaging types, etc.

# Classes of Interest

* `DependencySynonymFormatter`
* `CompatibleDependencyFormatRegistry`

# Existing Implementations For Maven Artifacts

The following could serve as examples of how this has been implemented for other build tools:
* `IvyDependencyFormatter` / `IvyDependencyFormatterTest`
* `GradleDependencyFormatter` / `GradleDependencyFormatterTest`
* `LeiningenDependencyFormatter` / `LeiningenDependencyFormatterTest`
* `MavenDependencyFormatter` / `MavenDependencyFormatterTest`
* `SBTDependencyFormatter` / `SBTDependencyFormatterTest`

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by Declan-Y**
Fixes #1270.

- Implemented BazelDependencyFormatter class
- Implemented BazelDependencyFormatterTest class
- Updated array in testRegisteredSynonyms.

### CODE REVIEW

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

There seems to be a failing test. Would you mind looking into it? Thanks! :)

Kind regards,

Martin

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/dependency/snippet/BazelDependencyFormatter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        MavenArtifactCoordinates coordinates = (MavenArtifactCoordinates) artifactCoordinates;
+        
+        return "maven_jar(" +"\n"+"    name ="+" "
+        +"\""+coordinates.getArtifactId()+"\""+","+
+        (coordinates.getArtifactId() != null && coordinates.getGroupId() != null && coordinates.getVersion() != null ? 
+        "\n    artifact = " + "\""+
+        coordinates.getGroupId()+":"+coordinates.getArtifactId()+":"+coordinates.getVersion()+"\""+",\n)\n": 
+        "\n)\n");
+    }
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/dependency/snippet/BazelDependencyFormatter.java]: Would you mind showing me a formatted dependency as the code above would print it? Thanks!

**Comment by Declan-Y** [comment]
Fixed the code that was failing the test :)

**Comment by carlspring** [comment]
Thanks! :)

If you could now just sign the ICLA and answer my question the code review, that would be great!

**Comment by Declan-Y** [comment]
The code would print
```
maven_jar(
    name = "maven-snippet",
    artifact = "org.carlspring.strongbox:maven-snippet:1.0",
)
```
But the artifact field is optional according to the Bazel format so it would also print

```
maven_jar(
    name = "maven-snippet",
)
```
If just coordinates.GetArtifactId is given.

**Comment by carlspring** [comment]
Thanks!

Would you mind rebasing?

Also, is the extra comma really necessary at the end?

**Comment by carlspring** [comment]
Ah, now I see this is so by design!...
Okay, if you could just rebase and resolve the merge conflicts, that would be great! Thanks! :)

**Comment by Declan-Y** [comment]
Resolved the merge conflict, should be all good now.

**Comment by carlspring** [comment]
Congratulations on your first pull request! Please, join our [chat channel](https://chat.carlspring.org/), if you'd like to find out what else you could help with! :)

**Pull request merged**